File: stacks.js
class Node {
    constructor(value){
        this.value = value;
        this.next = null;
    }
}

class Stack {
    constructor(){
        this.top = null;
        this.length = 0;
        this.botton = null;
    }

    peek(){
        if (this.top){
            return this.top.value;
        }
    }

    push(value){
        const node = new Node(value);
        if (this.length === 0){
            this.top = node;
            this.bottom = node;
            this.length ++;
        }
        else
        {
            const holdingPointer = this.top;
            this.top = node;
            this.top.next = holdingPointer;
            this.length ++; 
        }
        return this.top;
    }

    pop(){
        if (!this.top){
            return null;
        }
        else 
        { 
            if (this.length > 0){
            const holdingPointer = this.top;
            this.top = this.top.next;
            }

            if  (this.top === this.bottom){
                this.bottom = null;
            }
            this.length --; 
            return this.top;
        }
    }
}

let stack = new Stack();
stack.push(2);
stack.push(3);
stack.pop();
stack.pop();
console.log(stack.top, stack.bottom);
console.log(stack.length)

File: stacks.py
class Node:
    def __init__(self, value):
        self.node = {
            "value": value,
            "next": None 
        }

    def get(self):
        return self.node

class Stack:
    def __init__(self):
        self.top = None
        self.bottom = None
        self.length = 0

    def peek(self):
        if not self.top:
            return None
        else:
            return self.top["value"]


    def push(self, value):
        new_node = Node(value=value).get()
        if not self.top:
            self.top = new_node
            self.bottom = new_node
        else:
            new_node["next"] = self.top
            self.top = new_node
        self.length += 1
        return self.top

    def pop(self):
        if self.length == 0:
            return None
        else:
            if self.length == 1:
                self.top = None
                self.bottom = None
            else:
                head_pointer = self.top["next"]
                self.top = head_pointer
            self.length -= 1
            return self.top

if __name__ == "__main__":
    s = Stack()
    s.push(2)
    s.push(10)
    s.push("google")

    print(s.peek())
    s.pop()
    s.pop()
    s.push("facebook")

    print(s.top, s.bottom, s.length)